lib/pbench/test/unit/server/test_users_api.py

```diff
@@ -0,0 +1,48 @@
+from http import HTTPStatus
+
+from pbench.test.unit.server import DRB_USER_ID
+from pbench.test.unit.server.headertypes import HeaderTypes
+
+
+class Testuser:
+    def test_get_user(self, client, server_config, build_auth_header):
+        header_param = build_auth_header["header_param"]
+        expected_status = HTTPStatus.OK
+        if not HeaderTypes.is_valid(header_param):
+            expected_status = HTTPStatus.UNAUTHORIZED
+
+        response = client.get(
+            f"{server_config.rest_uri}/user/drb",
+            headers=build_auth_header["header"],
+        )
+        assert response.status_code == expected_status
+        if header_param is HeaderTypes.VALID_ADMIN:
+            assert response.json == {
+                "username": "drb",
+                "id": int(DRB_USER_ID),
```

Comment:
Huh; I hadn't really noticed that you're returning the row ID. I'm not sure that's a good idea, since it's completely irrelevant to the user identity or management.

Reply:
Only if the requesting user is an `admin` will the API return the row id. If the requesting user is the user itself then the API don't return the user row id.

Reply:
An admin has no more use for an internal row ID than anyone else: this just isn't something we should be exposing. Maybe this is a great argument for simply removing it and doing the join on the oidc_id.

Reply:
Alright, we can remove the `id` or replace the `id` column to store the `oidc_id` and remove the separate `oidc_id`. My only concern on that is if we have multiple identity providers and only keep the `oidc_id` in our internal table as a primary key we can not guarantee that the key is always going to be unique since they are externally generated (even though there is an extremely rare chance to have the 2 similar UUIDs from 2 separate providers but still). Having our own `id` guarantees the uniqueness of primary key.